reactive/kubernetes_master.py

```diff
@@ -209,27 +211,27 @@ def check_for_upgrade_needed():
     migrate_from_pre_snaps()
     maybe_install_kube_proxy()
     update_certificates()
-    add_rbac_roles()
     switch_auth_mode(forced=True)
 
-    # Basic auth is gone in 1.19; ensure any custom entries are added to
-    # known_tokens.csv. This updates existing or appends new entries.
-    if is_leader and not is_flag_set('kubernetes-master.basic-auth.migrated'):
-        basic_auth = '/root/cdk/basic_auth.csv'
-        with open(basic_auth, 'r') as f:
-            rows = list(csv.reader(f))
-
-        for row in rows:
-            try:
-                if row[0].startswith('#'):
-                    continue
-                else:
-                    setup_tokens(*row)
-            except IndexError:
-                pass
-        kubernetes_master.deprecate_auth_file(basic_auth)
-
-        set_flag('kubernetes-master.basic-auth.migrated')
+    # File-based auth is gone in 1.19; ensure any entries in basic_auth.csv are
+    # added to known_tokens.csv, and any known_tokens entries are created as secrets.
```

Comment:
oooof!  excellent catch.  luckily your future commentary will fix this -- that is, instead of namespacing secrets, we'll set them up with a custom type.